homeassistant/components/google_assistant/trait.py

```diff
@@ -901,31 +907,85 @@ class TemperatureControlTrait(_Trait):
 
     name = TRAIT_TEMPERATURE_CONTROL
 
+    commands = [
+        COMMAND_SET_TEMPERATURE,
+    ]
+
     @staticmethod
     def supported(domain, features, device_class, _):
         """Test if state is supported."""
         return (
             domain == sensor.DOMAIN
             and device_class == sensor.SensorDeviceClass.TEMPERATURE
+        ) or (
+            domain == water_heater.DOMAIN
+            and features & WaterHeaterEntityFeature.TARGET_TEMPERATURE
```

Comment:
Let's swap the order to match the order in `sync_attributes` below in `query_attributes`, please do that here too.